src/frontend/src/flows/recovery/recoveryWizard.ts

```diff
@@ -127,36 +134,125 @@ const addDeviceWarningTemplate = ({
 // TODO: Create the `addDeviceWarning` page and use it in `recoveryWizard` function.
 export const addDeviceWarningPage = renderPage(addDeviceWarningTemplate);
 
+// Prompt the user to create a recovery phrase
+export const addDeviceWarning = ({
+  status,
+}: {
+  status: DeviceStatus;
+}): Promise<{ action: "remind-later" | "do-not-remind" | "add-device" }> => {
+  return new Promise((resolve) =>
+    addDeviceWarningPage({
+      i18n: new I18n(),
+      ok: () => resolve({ action: "add-device" }),
+      remindLater: () => resolve({ action: "remind-later" }),
+      doNotRemindAgain: () => resolve({ action: "do-not-remind" }),
+      status,
+    })
+  );
+};
+
+/**
+ * Helper to encapsulate the logic of when and which recovery warning page to show.
+ *
+ * Three conditions must be met for the warning page to be shown:
+ * * Not having seen the recovery page in the last week
+ *    (on registration, the user is not shown the page, but set it as seen to not bother during the onboarding)
+ * * The user has less than one device.
```

Comment:
The user has at most one device, right ?

Reply:
Yes, correct. I rephrased it.